### PR TITLE
Adds toggle to be used with editing payment methods

### PR DESCRIPTION
**Pull request by belcherj**
Because the FormToggle component does not return an event when calling the onChange method I need to wrap it in a component that does.

This component mimics the event object structure so that it can be parsed by the same function as other inputs.

This can only be tested on its own since the code that requires this component is not part of this PR.  I have a PR up that has all of the code including the component that requires this here:  #14607

### CODE REVIEW

File: client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import FormToggle from 'components/forms/form-toggle';
+
+const PaymentMethodEditFormToggle = ( { checked, name, onChange } ) => {
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js]: Since we will probably have other WooCommerce settings that return 'yes'/'no' that a helper component would be useful for, should we make this name a little more general and put it in `woocommerce/components`?

File: client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			target: {
+				name,
+				value: checked ? 'no' : 'yes'
+			}
+		};
+		onChange( fakeEvent );
+	};
+
+	return (
+		<FormToggle
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js]: We should pass down any other passed props so that this FormToggle can support all other FormToggle props. 

Here is an example of what I mean: https://github.com/Automattic/wp-calypso/blob/c19d6fa3b512621130be715cf603094e519a9086/client/extensions/woocommerce/components/form-click-to-edit-input/index.js#L74

File: client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js (hunk 1 above)
Comment (belcherj) [inline client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js]: Should we do this when another component needs to use it?  Can I do this in another PR as several other PR's depend on this one?

File: client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js [hunk 3]
```diff
@@ -21,6 +23,7 @@ const PaymentMethodEditFormToggle = ( { checked, name, onChange } ) => {
 
 	return (
 		<FormToggle
+			{ ...omit( props, [ 'updateAriaLabel', 'editAriaLabel' ] ) }
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js]: Oops - the array here is any props you don't want passed along, which is probably just `onChange` since you set a handler below.

File: client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js (hunk 1 above)
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/payments/payment-method-edit-form-toggle.js]: Yeah -- that's fine with me. Maybe add a TODO in here about it?
